src/Main.hs

```diff
@@ -35,30 +37,31 @@ main = do
   let log = logMessage logR
   log "Starting up functorrent"
   log $ "Parsing input file " ++ concat args
-  parseTorrentFile log args >>= startTorrentConc log
+
+  case args of
+    [file] -> do
+              minfo <- parseTorrentFile log file
+              case minfo of
+                Right m -> startTorrentConc log m
+                Left err -> log err
+    [] -> usage
+
   logStop logR
 
 usage :: IO ()
-usage = putStrLn "usage: functorrent torrent-file"
+usage = putStrLn "Usage: functorrent torrent-file"
+
 
-parseTorrentFile :: Log -> [String] -> IO [Metainfo]
-parseTorrentFile log [a] = do
-  fileExist <- doesFileExist a
+parseTorrentFile :: Log -> String -> IO (Either String Metainfo)
```

Comment:
Let it be. We could decide on some logging guidelines and add logs inside the function.